src/App.jsx

```diff
@@ -14,70 +16,157 @@ export const goods = [
-          </td>
-
-          <td data-cy="GoodTitle" className="is-vcentered">
-            Dumplings
-          </td>
-        </tr>
-
-        <tr data-cy="Good" className="has-background-success-light">
-          <td>
-            <button
-              data-cy="RemoveButton"
-              type="button"
-              className="button is-info"
-            >
-              -
-            </button>
-          </td>
-
-          <td data-cy="GoodTitle" className="is-vcentered">
-            Jam
-          </td>
-        </tr>
-
-        <tr data-cy="Good">
-          <td>
-            <button
-              data-cy="AddButton"
-              type="button"
-              className="button"
-            >
-              +
-            </button>
-          </td>
-
-          <td data-cy="GoodTitle" className="is-vcentered">
-            Garlic
-          </td>
-        </tr>
-      </tbody>
-    </table>
-  </main>
-);
+export const ItemGood = (
+  {
+    good, selectedGood, setselectedGood,
+  },
+) => {
+  const [btnClickVerification, setBtnClVer] = useState(false);
+
+  return (
+    <tr data-cy="Good" className={`${btnClickVerification && 'has-background-success-light'}`}>
+      <td>
+        <button
+          data-cy={`${btnClickVerification ? 'RemoveButton' : 'AddButton'}`}
+          type="button"
+          className={cn('button', {
+            'is-info': btnClickVerification,
+          })}
+          onClick={
+            ({ type }) => {
+              if (type === 'click') {
+                setBtnClVer(true);
+                setselectedGood(`${good} is selected`);
+              }
+            }
+          }
+        >
+          {`${btnClickVerification ? '-' : '+'}`}
+        </button>
+      </td>
+
+      <td data-cy="GoodTitle" className="is-vcentered">
+        {good}
+      </td>
+    </tr>
+  );
+};
+
+export const App = () => {
+  const [selectedGood, setselectedGood] = useState('Jam is selected');
+  const [goodsMod] = useState(goods);
+
+  return (
+    <main className="section container">
+      <h1 className={
+        `title is-flex is-align-items-center ${selectedGood ? 'is-hidden' : ''}`
+        }
+      >
+        No goods selected
+      </h1>
+
+      <h1 className={
+        `title is-flex is-align-items-center ${!selectedGood ? 'is-hidden' : ''}`
+        }
+      >
+        {selectedGood}
+
+        <button
+          data-cy="ClearButton"
+          type="button"
+          className="delete ml-3"
+          onClick={
+            () => {
+              setselectedGood('');
+            }
+          }
+        />
+      </h1>
+
+      <table className="table">
+        <tbody>
+          {goodsMod.map(item => (
+            <ItemGood
+              key={item}
+              good={item}
+              selectedGood={selectedGood}
+              setselectedGood={setselectedGood}
+            />
+          ))}
+        </tbody>
+      </table>
+    </main>
+  );
+};
+
+export const App1 = () => {
```

Comment:
redundant component

Reply:
'Buttons should unclick to default state'
That is a ptoblem, I have no idea how to resolve that